romancal/dark_current/dark_sub.py

```diff
@@ -140,11 +141,16 @@ def average_dark_frames(input_dark, ngroups, nframes, groupgap):
         New dark object with averaged frames
 
     """
-
     # Create a model for the averaged dark data
     dny = input_dark.data.shape[1]
     dnx = input_dark.data.shape[2]
     avg_dark = datamodels.DarkModel((ngroups, dny, dnx))
+
+    # 03/12/21 this block is a workaround for datamodel's update bug
+    input_dark.meta.instrument.detector = 'WFI01'
+    avg_dark.meta.instrument.name = 'WFI'
+    avg_dark.meta.instrument.detector = 'WFI01'
+    avg_dark.meta.update(input_dark.meta._instance)
```

Comment:
What is the bug in datamodels and is there a ticket filed?

Reply:
Perhaps it is not technically a bug, but according to @PaulHuwe , the dictionary function 'update' has some issues with recursion, so the code modification here is his brute force implementation.

Reply:
To link to this PR, I have created this stdatamodel GH PR regarding the update needed in the datamodel  : "Improve datamodel dictionary 'update' handling of recursion" - (https://github.com/spacetelescope/stdatamodels/issues/66. ) I have been unable to link these 2 PRs, because of what may be a permission problem.